docs/2-fork-accountability.md

```diff
@@ -0,0 +1,191 @@
+
+In particular, we assume there are more than one third of faulty validators such that the following holds:
+
+    f < num. faulty processes <= 2f
+
+In the Tendermint protocol, all processes should be accountable for their actions and faulty validators should be identified and punished according to the protocol specifications, for example by removing them from the validators set.
+We aim to give incentives to validators to behave correctly according to the protocol specifications and detect faulty validators, without mistakenly detecting correct validators. 
+ 
+## Misbehaviors
+ 
+In the previous section, we mentioned that a process can make the consensus fail with a fork. There are, indeed, multiple reasons that can generate a fork in the blockchain, all coming from processes that deviate from the protocol specification in various ways.
+
+We can summarize some of the most important misbehaviors that can possibly cause a fork in a consensus instance according to the Tendermint consensus algorithm rules: 
+
+- A process sends multiple proposals in a round *r* for different values 
+
+- A process sends a PREVOTE/PRECOMMIT message for an invalid value *v* in a round *r*
+
+- A process sends multiple PREVOTE/PRECOMMIT messages in a round *r* for different values (**equivocation**)
+
+- A process sends a PREVOTE/PRECOMMIT message for a value *v* despite having a lock on a different value *v'*
+
+- A process sends a PRECOMMIT for a value *v* without having received at least two thirds PREVOTE messages for *v* 
+
+Processes are considered correct if they do not make any of the above-mentioned misbehaviors.
+
+## Design of an accountability algorithm
+
+We aim to develop an accountability algorithm that, by analyzing the messages received by validators during an execution of a consensus instance, is able to determine which processes respected the protocol and which ones did not and led to a fork. 
+In order to prove the misbehavior of the processes that did not respect the protocol, we also want to show the proof of their misbehavior with respect to the Tendermint consensus protocol.
+
+### Interface
+
+The algorithm exposes the following interface (in the event of a fork):
+
+- **detect(P)**: process P is faulty according to the accountability algorithm
+
+### Properties
+
+The algorithm guarantees the following properties:
+
+- **accuracy**: if the accountability algorithm detects a process *p*, then process *p* is faulty
+
+- **f+1-completeness**: the accountability algorithm eventually detects *f + 1* different processes
+
+All the processes that are not detected are considered correct.
```

Comment:
Maybe we should state here that some of the undetected processes can be faulty.
Just to make sure that reader understands that we can not detect all faulty processes.

Reply:
I agree, good suggestion. Slightly changed the sentence to make this clear.